### ISSUE TITLE
Map constructor doesn't work on destroy()

### ISSUE THREAD

**Issue by tomasbartalos** [grails/grails-core#10114]
If I try to use Map constructor on domain class after app destroy, the fields are not set (stays null). Happens only on domain classes, POJOs work fine.

### Steps to Reproduce
**Domain class:**
```class TestDestroy {
    String x
}```

**Service:**
```
class TestDestroyService implements DisposableBean {
    @Override
    void destroy() throws Exception {
        println("Destroying from disposable bean ....")
        TestDestroy objFromMap = new TestDestroy(x: "x")
        TestDestroy objSetter = new TestDestroy()
        objSetter.x = "x"
        TestDestroyPojo pojo = new TestDestroyPojo(x: "x")
    
        println("DisposableBean: Domain Map constructor has correct props: ${objFromMap.x == "x"}")
        println("DisposableBean: Domain setter has correct props: ${objSetter.x == "x"}")
        println("DisposableBean: Pojo has correct props: ${pojo.x == "x"}")
    }
}
```

When you shutdown the web application the output is:
> 2016-08-23 11:38:29.165:INFO:t.1:Destroying Spring FrameworkServlet 'grails'
> Destroying from disposable bean ....
> DisposableBean: Domain Map constructor has correct props: false
> DisposableBean: Domain setter has correct props: true
> DisposableBean: Pojo has correct props: true
> 2016-08-23 11:38:29.558:INFO:t.1:Closing Spring root WebApplicationContext

### Expected Behaviour

The output should be:
> DisposableBean: Domain Map constructor has correct props: true

Since the map constructor should set the field.

Note: when the destroy method is called from `Bootstrap.destroy{}`, it works fine. I could reproduce the problem only with Spring's DisposableBean interface.

### Environment Information

- **Grails Version:** 2.4.4
- **JDK Version:** 1.7.0_79
- **Container Version:** jetty-8.1.9

### Example Application

- link to github: https://github.com/tomasbartalos/grailsDestroyBug
- link to SO: http://stackoverflow.com/questions/39076268/map-constructor-doesnt-work-on-app-destroy

**Comment by jeffbrown** [comment]
I am not sure GORM is supposed to be accessible that late in the process.  There may be pieces that are already shut down by that point.  @graemerocher what say ye?

**Comment by graemerocher** [comment]
yes there are no guarantees on the order of destroy method calls. GORM might well be shutdown at this point

**Comment by tomasbartalos** [comment]
It worked seamlessly in grails 2.1.2 . After migrating to 2.4.4 we had really hard time to trace this down. When I set the property via setter `objSetter.x = "x"` it works and the object is persisted without any further errors. Which is so weird behavior, I would expect some session closed exception...
Seems like the Map constructor is the only broken part.
